test/WebJobs.Script.Tests.Integration/Host/HostIdValidatorTests.cs

```diff
@@ -127,10 +127,13 @@ public async Task ValidateHostIdUsageAsync_Collision_WarningLevel_Logs()
             await _hostIdValidator.ValidateHostIdUsageAsync(_testHostId);
 
             var logs = _loggerProvider.GetAllLogMessages();
-            var log = logs.Single();
+            var log = logs[0];
             Assert.Equal(LogLevel.Warning, log.Level);
             Assert.Equal(string.Format(Resources.HostIdCollisionFormat, _testHostId), log.FormattedMessage);
 
+            var diagLog = logs[1];
```

Comment:
Should we assert on `logs` length before accessing values?

Reply:
Seconding this suggestion.